sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update.hpp

```diff
@@ -475,40 +561,69 @@ struct ImplFcmpwr : ImplCmpxchgBase<T, N, atomic_op, atomic_op::fcmpxchg> {};
 
 // Main function and test combinations.
 
+template <bool UseAcc, class T, int N, template <class, int> class ImplF,
+          bool UseMask, bool UsePVCFeatures>
+auto run_test(queue q, const Config &cfg) {
+  if constexpr (UseAcc)
+    return test_acc<T, N, ImplF, UseMask, UsePVCFeatures>(q, cfg);
+  else
+    return test<T, N, ImplF, UseMask, UsePVCFeatures>(q, cfg);
+}
+
 template <int N, template <class, int> class Op, bool UseMask,
-          bool UsePVCFeatures, int SignMask = (Signed | Unsigned)>
+          bool UsePVCFeatures, bool UseAcc, int SignMask = (Signed | Unsigned)>
 bool test_int_types(queue q, const Config &cfg) {
   bool passed = true;
+#ifdef __ESIMD_FORCE_STATELESS_MEM
+  constexpr bool test_long = true;
+  constexpr bool test_short = true;
+#else
+  constexpr bool test_long = !UseAcc;
+  constexpr bool test_short = !UseAcc;
```

Comment:
Are short an long supported only in usm atomic update?

Reply:
yeah same as other comment about double, only 4 byte data is supported

Reply:
i just pushed a better fix, thanks